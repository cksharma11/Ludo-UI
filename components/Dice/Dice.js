import React from 'react';
import PropTypes from 'prop-types';
import DiceStyles from './Dice.style';
import diceImages from '../../config/images/dice-images';

const Dice = ({ diceValue }) => {
  return (
    <div>
      <img src={diceImages[diceValue]} alt="dice" id="dice" />
      <style jsx>{DiceStyles}</style>
    </div>
  );
};

Dice.defaultProps = {
  diceValue: 1
};

Dice.propTypes = {
  diceValue: PropTypes.number
};

export default Dice;
