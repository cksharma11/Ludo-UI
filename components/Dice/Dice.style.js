import css from 'styled-jsx/css';

const DiceStyles = css`
  img {
    height: 100px;
    width: 100px;
    margin-top: 45px;
  }
`;

export default DiceStyles;
