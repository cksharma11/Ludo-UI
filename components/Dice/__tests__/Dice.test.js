import React from 'react';
import { shallow } from 'enzyme';
import Dice from '../Dice';

describe('Dice', () => {
  it('should render properly', () => {
    const wrapper = shallow(<Dice />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
