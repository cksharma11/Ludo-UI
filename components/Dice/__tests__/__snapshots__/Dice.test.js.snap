// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dice should render properly 1`] = `
"<div className=\\"jsx-865041945\\">
  <img src=\\"https://upload.wikimedia.org/wikipedia/commons/0/09/Dice-1.svg\\" alt=\\"dice\\" id=\\"dice\\" className=\\"jsx-865041945\\" />
  <JSXStyle id=\\"865041945\\">
    i
    m
    g
    .
    j
    s
    x
    -
    8
    6
    5
    0
    4
    1
    9
    4
    5
    {
    h
    e
    i
    g
    h
    t
    :
    1
    0
    0
    p
    x
    ;
    w
    i
    d
    t
    h
    :
    1
    0
    0
    p
    x
    ;
    m
    a
    r
    g
    i
    n
    -
    t
    o
    p
    :
    4
    5
    p
    x
    ;
    }
  </JSXStyle>
</div>"
`;
